pkg/kudoctl/packages/reader/reader_tar.go

```diff
@@ -15,16 +16,19 @@ import (
 	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/convert"
 )
 
-func ReadTar(fs afero.Fs, path string) (*packages.Resources, error) {
+// ResourcesFromTar extracts files from the tar provides by he `inFs` and the `path` and converts
+// them to resources. All the extracted files are saved in the `outFs` for later use (e.g. searching
+// for local dependencies)
+func ResourcesFromTar(in afero.Fs, out afero.Fs, path string) (*packages.Resources, error) {
```

Comment:
Well... In this case I'm not opposed to passing in the output FS - It would allow to extract multiple tar files into the same MemFS, or into the real Filesystem, etc.

I'm not sure about the Naming though - `ExtractTarIntoFs` might be more fitting. I'll have to look into it :)

Reply:
I went back and forth on passing `out` as a parameter or simply returning it. In the end, I decided **to pass it,** with the following rationale: the caller of this method might very well decide to pass in a different kind of `Fs` and frankly, this should be ok 🤷

Reply:
I agree that this is the best way to abstract away the filesystem used in these calls. However, we only use it with `afero.MemMapFs` and don't need this, IMO. But I don't feel strongly about this. Though this makes the function signatures a bit harder to parse. Great, that `out` is used as parameter name to clearly indicate the parameters purpose.